compaction/compaction.cc

```diff
@@ -418,9 +418,12 @@ struct compaction_read_monitor_generator final : public read_monitor_generator {
 
 class formatted_sstables_list {
     bool _include_origin = true;
-    std::vector<sstring> _ssts;
+    std::vector<std::string> _ssts;
```

Comment:
Why not to use `std::list` or `std::deque` or alike? 
`std::vector` is a wrong container if you expect a frequent (more than 1) size change.

Reply:
linked list is overkill, we know the size upfront, we don't have to reallocate. the container size rarely go beyond 1k, so the memory usage is bounded to ~100k bytes. we could have picked chunked_vector, but don't think it's a must.

Reply:
1k -> 8kB

Reply:
> 1k -> 8kB

I was assuming ~100 bytes for each entry, as it duplicates the directory full path for each one of them.

Reply:
Ah dismiss what I said. I forget that std::string switch to heap once the string becomes big enough. Usually the stack limit is 32 bytes

Reply:
> linked list is overkill,

Overkill for what? What are you trying to optimize here exactly? 
If a container is going to allocate 15Bx1K == 15KB it's an unnecessary pressure on the seastar memory allocator that would have to find such a large contiguous buffer. Why not to use a `chunked_vector` or a `deque` and not bother about it at all?

Reply:
we haven't seen any need to optimize this particular variable.
`chunked_vector` could indeed do a good job.
Feel free to send a patch for master (I doubt it'd be backported though)

Reply:
Ah, and regarding vector vs. list; we do prefer a single allocation than yet another allocation per sstable.
Hence chunked_vector being a good compromise between a vector and a list.

Reply:
it indeed doesn't hurt to switch to chunked_vector and it can handle extreme unexpected cases.

Reply:
cc https://github.com/scylladb/scylladb/issues/14264